docs/source/catalog.rst

```diff
@@ -136,11 +131,88 @@ returned data.  Each data source has several attributes:
-a data source might accept a "postal_code" argument which is used to alter a database query, or select a particular
-group within a file.  Users set parameters with keyword arguments to the ``get()`` method on the catalog object.
+
+Templating
+''''''''''
+
+Intake catalog files support Jinja2 templating for driver arguments. Any occurrence of
+a substring like ``{{field}}`` will be replaced by the value of the user parameters with
+that same name, or explicitly provided by the user. For how to specify these user parameters,
+see the next section.
+
+Some additional values are available for templating. The following is always available:
+
+- ``CATALOG_DIR``: The full path to the directory containing the YAML catalog file.  This is especially useful
+  for constructing paths relative to the catalog directory to locate data files and custom drivers.
+
+so that the search for CSV files for the two "entry1" blocks, above, will happen in the same directory as
+where the catalog file was found.
+
+The following functions `may` be available. Since these execute code, the user of a catalog may decide
+whether they trust those functions or not.
+
+- ``env("USER")``: look in the environment for the named variable
+- ``client_env("USER")``: exactly the same, except that when using a client-server topology, the
+  value will come from the environment of the client.
+- ``shell("get_login thisuser -t")``: execute the command, and use the output as the value. The
+  output will be trimmed of any trailing whitespace.
+- ``client_shell("get_login thisuser -t")``: exactly the same, except that when using a client-server
+  topology, the value will come from the system of the client.
+
+The reason for the "client" versions of the functions is to prevent leakage of potentially sensitive
+information between client and server by controlling where lookups happen. When working without a server,
+only the ones without "client" are used.
+
+An example:
+
+.. code-block:: yaml
+
+    sources:
+      personal_source:
+        description: This source needs your username
+        args:
+          url: "http://server:port/user/{{env(USER)}}"
+
+Here, if the user is named "blogs", the ``url`` argument will resolve to
+``"http://server:port/user/blogs"``; if the environment variable is not defined, it will
+resolve to ``"http://server:port/user/"``
+
+.. _paramdefs:
+
+Parameter Definition
+''''''''''''''''''''
+
+A source definition can contain a "parameters" block.
+A parameter may look as follows:
+
+.. code-block:: yaml
+
+    parameters:
+      name:
+        description: human-readable text for what this parameter means
+        type: optional, one of bool, str, int, float, list[str], list[int], list[float], datetime
+        default: optional, value to assume if user does not override
+        allowed: optional, list of values that are OK, for validation
+        min: optional, minimum allowed, for validation
+        max: optional, maximum allowed, for validation
```

Comment:
This section is slightly confusing because it is formatted like an example but it isn't an example. Could the explanatory text be in comments?